interfaces/builtin/audit_control.go

```diff
@@ -0,0 +1,70 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const auditControlSummary = `allows control over the kernel audit system`
+
+const auditControlBaseDeclarationSlots = `
+  audit-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const auditControlConnectedPlugSecComp = `
+# Description: Can use netlink to communicate with kernel audit system.
+bind
+socket AF_NETLINK - NETLINK_AUDIT
+`
+
+const auditControlConnectedPlugAppArmor = `
+# Description: Can use netlink to communicate with kernel audit system.
+network netlink,
+
+# CAP_AUDIT_CONTROL required to enable/disable kernel auditing, change auditing
+# filter rules, and retrieve auditing status and filtering rules.
+capability audit_control,
```

Comment:
In the netlink_audit interface we check that the apparmor sandbox has the required cap-audit-read parser feature (and hence that AppArmor appears to support `capability audit_read` etc - do we think we need a similar check here? In the netlink_audit interface this apparently is because AppArmor on 14.04 doesn't support this capability - so I think this new interface should include a similar probe etc to keep the two consistent.

Reply:
Actually I don't think this is needed - looking into it a bit more, CAP_AUDIT_CONTROL has existed a lot longer (since 2.6.11) than CAP_AUDIT_READ (3.16) and regardless, apparmor 2.10.95 as shipped in Ubuntu 14.04 supports both `capability audit_read`  and `capability audit_control`.

Reply:
Right, that was my understanding as well, given that there are no checks for parser support for `cap audit_write`, and `audit_control` has existed as long as `audit_write`.